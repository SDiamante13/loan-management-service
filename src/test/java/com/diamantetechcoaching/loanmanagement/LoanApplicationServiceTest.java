package com.diamantetechcoaching.loanmanagement;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

@SpringBootTest
class LoanApplicationServiceTest {

    @Autowired
    LoanApplicationService loanApplicationService;

    @Test
    void rejectedLoans() {
        LoanApplicationResponse lowCreditScore = loanApplicationService.processLoanApplication(new LoanApplicationRequest(
                "Bruce",
                "Banner",
                5_000,
                1_500,
                18_000,
                "999"
        ), 700, loanEntity -> {
        });

        Approvals.verify(lowCreditScore);
    }

    @Test
    void approvedLoans() {
        LoanApplicationResponse approvedLoan = loanApplicationService.processLoanApplication(new LoanApplicationRequest(
                "Bruce",
                "Banner",
                5_000,
                1_500,
                18_000,
                "999"
        ), 750, loanEntity -> {
        });

        Approvals.verify(approvedLoan);
    }
}