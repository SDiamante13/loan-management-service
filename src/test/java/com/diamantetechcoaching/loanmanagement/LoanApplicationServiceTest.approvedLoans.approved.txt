LoanApplicationResponse{status='Approved', creditScore=750, monthlyIncome=5000.0, monthlyDebt=1500.0, requestedAmount=18000.0, dti=30.0}